packages/relay-compiler/core/RelayFileIRParser.js

```diff
@@ -71,11 +70,46 @@ function parseFile(file: string): ?DocumentNode {
   };
 }
 
-function getParser(baseDir: string): FileParser {
-  return new FileParser({
-    baseDir,
-    parse: parseFile,
-  });
+type TransformFactory = (baseDir: string) => (filename: string, text: string) => string
+type TransformModule = { default: TransformFactory }
+
+function getParser(transformModules: Array<string> = []) { 
+  return (baseDir: string): FileParser => {
+    const transformer = getTransformer(baseDir, transformModules)
+    return new FileParser({
+      baseDir,
+      parse: (filename: string) => {
+        const text = fs.readFileSync(filename, 'utf8');
+        return parseFile(filename, transformer(filename, text))
+      },
+    });
+  }
+}
+
+function getTransformer(baseDir: string, transformModules: Array<string> = []) {
+  let transformer = (filename: string, text: string) => text 
+  if (transformModules.length) {
+    transformModules.forEach(moduleName => {
+      let moduleImpl: TransformFactory
+      try {
+        // $FlowFixMe flow doesn't know about __non_webpack_require__
+        moduleImpl = (__non_webpack_require__(moduleName): TransformFactory)
```

Comment:
Probably. I need to re-test my PR. If it doesn't work can try this "one weird trick".